### PR TITLE
Preserve formatted number on handleFlagItemClick (#185)

### LINKED ISSUE #185: Switching to a different country causing the phone number value to lose the + sign
If I type the numbers in, the value assigned to the phone value is correct, like +1 111 111
However if I change the country code from the dropdown list, without editing the phone  number, say the country code is +54, value I got become 54 111 111. I lose the + sign. 
If I edit it again, the + sign come back.

In this case I had to manually put the + sign back in which is ugly.

### PR DESCRIPTION
**Pull request by fleetofthemalden**
### Description
- pass event object through handleFlagItemClick
- pass formattedNumber to onChange handler

### Links
fixes #185

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -537,7 +537,7 @@ class PhoneInput extends React.Component {
       formattedNumber
     }, () => {
       this.cursorToEnd();
-      if (this.props.onChange) this.props.onChange(formattedNumber.replace(/[^0-9]+/g,''), this.getCountryData());
```
Comment (fleetofthemalden) [inline src/index.js:540]: this regex replace was turning this into an unformatted number. Was that just an artifact from a previous implementation? Or was the intention something I am not seeing?

In handleInput it uses the formattedNumber https://github.com/bl00mber/react-phone-input-2/pull/205/files#diff-1fdf421c05c1140f6d71444ea2b27638L510

**Comment by borzecki** [comment]
@bl00mber when you change value in input it returns valid number with `+` as prefix. In case of changing the country it get's removed leading to inconsistency.

**Comment by fleetofthemalden** [comment]
> The callback should give the user unformatted value. Why user will ever need formatted one?

`onChange`[ passes a formatted number.](https://github.com/bl00mber/react-phone-input-2/blob/master/src/index.js#L524) All of the other handlers pass formatted numbers through the event object. `handleFlagItemClick` is unique in that it returns unformatted, hence the inconsistency. If you pass a focus or blur handler, the [value attached to the event](https://github.com/bl00mber/react-phone-input-2/blob/master/src/index.js#L849) is going to be formatted as well.

I selected this package because it returns formatted numbers. Our project relies on formatted numbers. I can see scenarios where a user would want a raw number, however it's fairly trivial to strip formatting (like with your regex), but it is non-trivial to add it to the returned value.

Because the formatted number exists on the event object, to fix the underlying inconsistency we can
1) Continue to pass the event handler as done in this fix (because that way one can still grab the formatted number from the `event.value`), but [modify the changeHandler](https://github.com/bl00mber/react-phone-input-2/blob/master/src/index.js#L524) to pass the unformatted number instead. I suspect this would break projects for a lot of existing users.
2) Merge this PR as is.

@bl00mber  Please let me know how you would like to proceed.

**Comment by bl00mber** [comment]
It should pass unformatted number. As I've questioned above, why user will ever need formatted value? So I changed the method in 7ade064c9ff9928e576770132312fb6a87922320 and now it is passing an unformatted value. Thanks for the PR!

**Comment by fleetofthemalden** [comment]
> It should pass unformatted number. As I've questioned above, why user will ever need formatted value? So I changed the method in [fffd865](https://github.com/bl00mber/react-phone-input-2/commit/fffd865f01c47588fbf202a885db2857d397370f) and now it passing an unformatted value. Thanks for the PR!

Some databases, including the one for the enterprise trucking software that I work on, require a formatted phone number. So yes there is production software that relies on it. We could do the formatting on the back-end, but if it is already formatted on the front-end, why do the extra work? Since it is available via the event object, I can still use this library.

**Comment by bl00mber** [comment]
Seems reasonable. I've added unformatted value as 4th argument in 1380e07e1304377fa4e3ec7f7aa1d8918c6f826d

**Pull request closed without merging**